modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -458,59 +473,65 @@ private Predicate _getInPredicate(
 		);
 	}
 
-	private EntityModel _getObjectDefinitionEntityModel(
-		long objectDefinitionId) {
+	private List<ObjectValuePair<ObjectRelationship, Long>>
+		_getListObjectRelationshipsRelatedObjectDefinitionId(
+			long objectDefinitionId, List<String> objectRelationshipsNames) {
 
-		EntityModel entityModel = _entityModels.get(objectDefinitionId);
+		List<ObjectValuePair<ObjectRelationship, Long>> objectValuePairs =
+			new ArrayList<>();
 
-		if (entityModel == null) {
-			entityModel = _createEntityModel(objectDefinitionId);
+		for (String objectRelationshipsName : objectRelationshipsNames) {
+			ObjectRelationship objectRelationship = _fetchObjectRelationship(
+				objectDefinitionId, objectRelationshipsName);
 
-			_entityModels.put(objectDefinitionId, entityModel);
-		}
+			if (objectRelationship != null) {
+				objectValuePairs.add(
+					new ObjectValuePair<>(
+						objectRelationship, objectDefinitionId));
 
-		return entityModel;
-	}
+				objectDefinitionId = _getRelatedObjectDefinitionId(
+					objectDefinitionId, objectRelationship);
+			}
+		}
 
-	private Predicate _getObjectRelationshipPredicate(
-		Object left,
-		UnsafeBiFunction<String, Long, Predicate, Exception> unsafeBiFunction) {
+		if (objectValuePairs.isEmpty()) {
+			throw new ServerErrorException(
+				500,
+				new Exception(
+					"Unexpected error. ObjectRelationships not founded"));
```

Comment:
```"Unexpected error. ObjectRelationships not found"));```

Reply:
I would leave as this because not finding object relationships at this point is strange. If the object relationship name is badly formed on the filter, the filter parser will jump with the error. At this point, the naming has been validated and not finding the relationship is an unexpected error.